bcl2fastq/lib/bcl2fastq_utils.py

```diff
@@ -41,9 +45,16 @@ def __init__(self,
         """
 
         self.runfolder_input = runfolder_input
-        self.samplesheet_file = runfolder_input + "/SampleSheet.csv"
         self.base_calls_input = runfolder_input + "/Data/Intensities/BaseCalls"
 
+        if not samplesheet:
```

Comment:
I'm thinking that since we don't have any way of recovering from the type of error caused by not having writing permissions we might as well fail with an exception...

When it comes to validating the samplesheet string I'm not sure we want to go there since it's such a complex process. Steps downstream of this that read the samplesheet will do minimal validation, and then bcl2fastq should exit with a non-zero value if it has a non-valid syntax.